bindings/java/src/org/sleuthkit/datamodel/timeline/TimelineFilter.java

```diff
@@ -410,10 +419,66 @@ public RootFilter copyOf() {
 
 		}
 
-		private boolean isNamedSubFilter(TimelineFilter subFilter) {
+		private boolean isNotNamedSubFilter(TimelineFilter subFilter) {
 			return !(namedSubFilters.contains(subFilter));
 		}
 
+		@Override
+		public String toString() {
+			return "RootFilter{" + "knownFilter=" + knownFilter + ", tagsFilter=" + tagsFilter + ", hashFilter=" + hashFilter + ", textFilter=" + textFilter + ", typeFilter=" + typeFilter + ", dataSourcesFilter=" + dataSourcesFilter + ", fileTypesFilter=" + fileTypesFilter + ", namedSubFilters=" + namedSubFilters + '}';
+		}
+
+		@Override
+		public int hashCode() {
+			int hash = 7;
+			hash = 17 * hash + Objects.hashCode(this.knownFilter);
+			hash = 17 * hash + Objects.hashCode(this.tagsFilter);
+			hash = 17 * hash + Objects.hashCode(this.hashFilter);
+			hash = 17 * hash + Objects.hashCode(this.textFilter);
+			hash = 17 * hash + Objects.hashCode(this.typeFilter);
+			hash = 17 * hash + Objects.hashCode(this.dataSourcesFilter);
+			hash = 17 * hash + Objects.hashCode(this.fileTypesFilter);
+			hash = 17 * hash + Objects.hashCode(this.namedSubFilters);
+			return hash;
+		}
+
+		@Override
```

Comment:
Is there away to make hashCode and equals more dynamic so that if other xxxFilter are added that these two functions may not have to be revisited?  Can you use nameSubFilters to loop through the filters so that you don't have to call out each one individually?